### DISCUSSION TITLE
SpecRun plugin doesn't work with Reqnroll. couldn't configure maximum number of threads nor retries!

### DISCUSSION THREAD

**Discussion by AhmedElbaik** [reqnroll/Reqnroll#166] [Ideas]
I went through the documentation, and I found nothing related to how to configure the maximum number of threads and the retries.
Using Reqnroll.Xunit

**Comment by gasparnagy** [comment]
Unfortunately SpecRun is a closed source discontinued product, so we could not port that to Reqnroll. See https://reqnroll.net/news/2024/02/from-specflow-to-reqnroll-why-and-how/#has-everything-been-forked-from-specflow for details.

**Reply by AhmedElbaik** [reply]
Thank you for your response. Do you have any suggestions for how I might implement retry for failed tests or how I can configure the number of threads in the meantime?!

**Reply by gasparnagy** [reply]
@AhmedElbaik I personally use the VSTest build task in Azure DevOps pipelines, that has a built-in re-run feature.

But there are couple of general options you can use, like the [dotnet-test-rerun](https://github.com/joaoopereira/dotnet-test-rerun) tool or [this PowerShell script](https://conductofcode.io/post/retry-flaky-tests-with-dotnet-test-and-powershell/). 

You can also consider to make a Reqnroll plugin for the same, but I think it is better to handle this "outside" of Reqnroll. The POC for a step-level retry plugin of @adrichem might be a good start. Check out the related discussion here: https://github.com/orgs/reqnroll/discussions/92#discussioncomment-9075538

**Comment by gasparnagy** [comment]
I close this discussion now, if you need further help on test re-run, let's continue it on a separate thread.
